src/sema/builtin.rs

```diff
@@ -881,54 +882,54 @@ pub fn resolve_call(
     context: &ExprContext,
     ns: &mut Namespace,
     symtable: &mut Symtable,
-    diagnostics: &mut Vec<Diagnostic>,
+    diagnostics: &mut Diagnostics,
 ) -> Result<Expression, ()> {
-    let matches = BUILTIN_FUNCTIONS
+    let funcs = BUILTIN_FUNCTIONS
         .iter()
         .filter(|p| p.name == id && p.namespace == namespace && p.method.is_none())
         .collect::<Vec<&Prototype>>();
+    let mut errors: Diagnostics = Default::default();
```

Comment:
If you derive `Default` for `Diagnostics`, you can use `Diagnostics::default()`, instead. This makes the code more readable.

Reply:
`Default` is derived for `Diagnostics`, that's why `Default::default()` works.

`Diagnostics::default()` is a nice compromise. I did have a `Diagnostics::new()` which just called `Default::default()` but I've removed that now, and I'm using `Diagnostics::default()` everywhere now.